test/e2e/helpers.go

```diff
@@ -0,0 +1,188 @@
+// asserting there were no errors.
+//
+// By default the call is retried for 5s with a 250ms interval.
+func ListK8sObjectWithRetry[T kruntime.Object](
+	ctx context.Context, list K8sListFunc[T], options metav1.ListOptions,
+) T {
+	var object T
+	Eventually(func(g Gomega, ctx context.Context) {
+		result, err := list(ctx, options)
+		g.Expect(err).NotTo(HaveOccurred())
+		object = result
+	}).WithContext(ctx).WithTimeout(DefaultTimeout).WithPolling(PollingInterval).Should(Succeed())
+	return object
+}
+
+// This function takes a create function like clients.Kubernetes.CoreV1().Pods(namespace).Create
+// and the parameters for it. It then makes the call with some retry logic and returns the result after
+// asserting there were no errors.
+//
+// By default the call is retried for 5s with a 250ms interval.
+func CreateK8sObjectWithRetry[T kruntime.Object](
+	ctx context.Context, create K8sCreateFunc[T], obj T, options metav1.CreateOptions,
+) T {
+	var object T
+	Eventually(func(g Gomega, ctx context.Context) {
+		result, err := create(ctx, obj, options)
+		g.Expect(err).NotTo(HaveOccurred())
+		object = result
+	}).WithContext(ctx).WithTimeout(DefaultTimeout).WithPolling(PollingInterval).Should(Succeed())
+	return object
+}
+
+// This function takes a delete function like clients.Kubernetes.CertificatesV1().CertificateSigningRequests().Delete
+// and the parameters for it. It then makes the call with some retry logic.
+//
+// By default the call is retried for 5s with a 250ms interval.
+func DeleteK8sObjectWithRetry(
+	ctx context.Context, delete K8sDeleteFunc, name string, options metav1.DeleteOptions,
+) {
+	Eventually(func(g Gomega, ctx context.Context) {
+		err := delete(ctx, name, options)
+		g.Expect(err).NotTo(HaveOccurred())
+	}).WithContext(ctx).WithTimeout(DefaultTimeout).WithPolling(PollingInterval).Should(Succeed())
+}
+
+type Project struct {
+	projectClient projectclient.Interface
+	cli           kubernetes.Interface
+	Name          string
+}
+
+func NewProject(
+	ctx context.Context, cli kubernetes.Interface, projectClient projectclient.Interface, name string,
+) Project {
+	p := Project{
+		projectClient: projectClient,
+		cli:           cli,
+		Name:          name,
+	}
+	createCall := p.projectClient.ProjectV1().Projects().Create
+	CreateK8sObjectWithRetry(ctx, createCall, &projectv1.Project{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: p.Name,
+		},
+	}, metav1.CreateOptions{})
+	return p
+}
+
+func (p Project) CleanUp(ctx context.Context) {
+	p.Delete(ctx)
+	p.VerifyProjectIsDeleted(ctx)
+}
+
+func (p Project) Delete(ctx context.Context) {
+	deleteCall := p.projectClient.ProjectV1().Projects().Delete
+	DeleteK8sObjectWithRetry(ctx, deleteCall, p.Name, metav1.DeleteOptions{})
+}
+
+// VerifyProjectIsReady verifies that the project and relevant resources have been created correctly and returns error otherwise
+func (p Project) Verify(ctx context.Context) error {
```

Comment:
Outdated comment or method name (VerifyProjectIsReady vs Verify).

 Also, I would say ..."and returns an error if any" as maybe the creation can be successful but other errors can arise.